### PR TITLE
Manual dictionary search

### PR DESCRIPTION
**Pull request by chelobaka**
This PR implements manual dictionary search. Automatic search can be switched on/off in the options. The feature is discussed here https://sourceforge.net/p/omegat/feature-requests/1401/.

### CODE REVIEW

File: src/org/omegat/gui/dictionaries/DictionariesTextArea.java [hunk 1]
```diff
@@ -325,6 +338,59 @@ public DictionaryEntriesSearchThread(final SourceTextEntry newEntry) {
         }
     }
 
+    /**
+     * Thread for user requested dictionary search.
+     */
+    public class DictionaryTextSearchThread extends Thread {
```
Comment (amake) [inline src/org/omegat/gui/dictionaries/DictionariesTextArea.java]: Is there a reason you didn't extend `EntryInfoSearchThread`? It handles error logging and edge cases like when the current entry has changed before the results come back.

Comment (amake) [reply src/org/omegat/gui/dictionaries/DictionariesTextArea.java]: (See `DictionaryEntriesSearchThread` for an example)

File: src/org/omegat/Bundle.properties [hunk 2]
```diff
@@ -441,6 +441,8 @@ TF_OPTIONSMENU_DICTIONARY=Dictionar&y
 TF_OPTIONSMENU_DICTIONARY_FUZZY=Use &Fuzzy Matching for Dictionary Entries
 PREFS_TITLE_DICTIONARY=Dictionary
 PREFS_DICTIONARY_FUZZY=Use &fuzzy matching for dictionary entries
+PREFS_DICTIONARY_AUTO_SEARCH=&Automatically search segment text in dictionary
+TF_MENU_EDIT_SEARCH_DICTIONARY=Search dictionaries
```
Comment (amake) [inline src/org/omegat/Bundle.properties]: A minor point, but menu commands like this should be in Title Case.

Comment (amake) [reply src/org/omegat/Bundle.properties]: Actually I'm going to fix that for you because I want to test our Azure CI integration.

**Comment by chelobaka** [comment]
@amake can you get back to this PR when you have time please?

**Comment by amake** [comment]
@chelobaka I was waiting for you to answer [my question](https://github.com/omegat-org/omegat/pull/27#discussion_r237722595) about the thread class (as I mentioned in my last comment).

**Comment by amake** [comment]
I've committed to trunk. Thanks for your contribution!

**Pull request closed without merging**